proxy/context.go

```diff
@@ -15,48 +15,111 @@
 package proxy
 
 import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
 	"time"
 
+	"github.com/golang/glog"
 	"github.com/golang/protobuf/ptypes"
 	proxyconfig "istio.io/api/proxy/v1/config"
 	"istio.io/pilot/model"
 )
 
-// Context defines local proxy context information about the global service mesh
-type Context struct {
+// Environment provides an aggregate environmental API for Pilot
+type Environment struct {
 	// Discovery interface for listing services and instances
-	Discovery model.ServiceDiscovery
+	model.ServiceDiscovery
 
 	// Accounts interface for listing service accounts
-	Accounts model.ServiceAccounts
+	model.ServiceAccounts
 
 	// Config interface for listing routing rules
-	Config model.IstioConfigStore
+	model.IstioConfigStore
 
-	// MeshConfig defines global configuration settings
-	MeshConfig *proxyconfig.ProxyMeshConfig
+	// Mesh is the mesh config (to be merged into the config store)
+	Mesh *proxyconfig.ProxyMeshConfig
+}
+
+// Role declares the proxy node role in the mesh
+type Role interface {
+	// nolint: megacheck
+	isProxyRole()
+
+	// ServiceNode encodes the role information into a string
+	ServiceNode() string
+}
 
+// Sidecar defines the sidecar proxy role
+type Sidecar struct {
 	// IPAddress is the IP address of the proxy used to identify it and its
 	// co-located service instances. Example: "10.60.1.6"
 	IPAddress string
 
-	// UID is the platform specific unique identifier of the proxy.
-	// UID should serve as a key to lookup additional information associated with the
-	// proxy. Example: "kubernetes://my-pod.my-namespace"
-	UID string
-
-	// PassthroughPorts is a list of ports on the proxy IP address that must be
-	// open and allowed through the proxy to the co-located service instances.
-	// These ports are utilized by the underlying cluster platform for health
-	// checking, for example.
-	//
-	// The passthrough ports should be exposed irrespective of the services
-	// model. In case there is an overlap, that is the port is utilized by a
-	// service for a service instance and is also present in this list, the
-	// service model declaration takes precedence. That means any protocol
-	// upgrade (such as utilizng TLS for proxy-to-proxy traffic) will be applied
-	// to the passthrough port.
-	PassthroughPorts []int
+	// PodName for the proxy sidecar
+	PodName string
+
+	// PodNamespace for the proxy sidecar
+	PodNamespace string
```

Comment:
Right but can't we concatenate the env vars and create a single podname string to be passed to proxy/ ? 
Just concerned that we are introducing minor semantics (components of pod fqdn)